### ISSUE TITLE
Can we use provisional notifications with React Native SDK?

### ISSUE THREAD

**Issue by raquelmsmith** [urbanairship/react-native-module#388]
# Preliminary Info

### What Airship dependencies are you using?

13.1.1

In the iOS documentation I see support for provisional authorization: https://docs.airship.com/platform/ios/push-notifications/#provisional-authorization

However, there is no mention of this in the React Native docs, and I don't see anything in the codebase here. Is provisional authorization supported in the React Native module?

Thank you!!

**Comment by raquelmsmith** [comment]
OK I'm actually seeing a method for
```
      UrbanAirship.setForegroundPresentationOptions({'alert': true, 'badge': true, 'sound': true})
```

Looks like it doesn't accept 'provisional.' Am I understanding that correctly? Can that be added?

**Comment by rlepinski** [comment]
Yep, we are going to work on a release this week. Hopefully will be out EOW but it might go into next week.

**Comment by rlepinski** [comment]
I have work started here - https://github.com/urbanairship/react-native-module/pull/397

Unfortunately some of types are poorly named so its making it a bit harder then it should be. The PR will add:
- ability to set notification options (including provisional)
- ability to get current authorization status
- expose full set of authorized settings in both the getter and the event
